modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaFieldWidget.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_featured_media_field\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\EntityReferenceAutocompleteWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Plugin implementation of the 'oe_featured_media_widget' widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_featured_media_widget",
+ *   label = @Translation("Featured media"),
+ *   description = @Translation("An autocomplete entity reference field and a text field."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaFieldWidget extends EntityReferenceAutocompleteWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $referenced_entities = $items->referencedEntities();
```

Comment:
No need to redo all the form, the parent class already does it. Do something like this:
`$element['featured_media'] = [`
`      '#type' => 'fieldset',`
  `    '#title' => $this->t('Featured media'),`
`    ] + parent::formElement($items, $delta, $element, $form, $form_state);`
  `  $element['featured_media']['caption'] = [`
`      '#title' => $this->t('Caption'),`
  `    '#type' => 'textarea',`
`      '#default_value' => $items[$delta]->caption,`
  `    '#rows' => 2,`
`    ];`

    return $element;`

Reply:
We have custom title, we have to make the help text work, and we need to place the field in a different element. In the parent method there is the autocreate entity we don't need as well.

Reply:
Plus when we use multi element field, it did not played out as I wanted due to the parent execution.

Reply:
Ok then keep it as before.